cmd/vela-worker/exec.go

```diff
@@ -70,6 +71,25 @@ func (w *Worker) exec(index int) error {
 		"version":  v.Semantic(),
 	})
 
+	// handle stale item queued before a Vela upgrade or downgrade.
+	if item.ItemVersion != types.ItemVersion {
+		// If the ItemVersion is older or newer than what we expect, then it might
+		// not be safe to process the build. Fail the build and loop to the next item.
+		// TODO: Ask the server to re-compile and requeue the build instead of failing it.
+		logrus.Errorf("Failing stale queued build due to wrong item version: want %d, got %d", types.ItemVersion, item.ItemVersion)
+
+		build := item.Build
+		build.SetStatus(constants.StatusFailure)
+
```

Comment:
should we set a nice, user friendly message on the build also?

Reply:
Oh. Excellent idea. Done!